### PR TITLE
Add connectivity

### LINKED ISSUE #181: Update connectivity plugin
As I know Android & iOS have new features to detect internet connection state. So we need review and update our implementation for connectivity.

### PR DESCRIPTION
**Pull request by sleushunou**
### Issues Resolved ### 

- fixes #181 

### API Changes ###

Added:
 - `Connectivity` projects

### Platforms Affected ### 

- Core (all platforms)
- iOS
- Android

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+            return types;
+        }
+
+        private NWPathMonitor CreateMonitor(NWInterfaceType type, Action<NWPath> action)
+        {
+            _connectionStatuses.Add(type, false);
+
+            var monitor = new NWPathMonitor(type);
+            monitor.SetQueue(CoreFoundation.DispatchQueue.MainQueue);
```
Comment (wcoder) [inline Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: Avoid using main thread:
`DispatchQueue.GetGlobalQueue(DispatchQueuePriority.Background)`

File: Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        ~IosConnectivityService()
+        {
+            Dispose(false);
+        }
+
+        protected void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _ceccularMonitor.Dispose();
```
Comment (wcoder) [inline Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: I think we need to call `.Cancel()` explicitly.

**Comment by wcoder** [comment]
Related: https://github.com/xamarin/Essentials/issues/633

File: Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs (hunk 1 above)
Comment (sleushunou) [reply Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: Background thread throws an exception

Comment (sleushunou) [reply Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: =================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x10289d2c8 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028930b0 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028a1998 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : mono_pmip
	0x21ed069fc - /usr/lib/system/libsystem_platform.dylib : <redacted>
	0x1028921a4 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028921a4 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028915c8 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x102887140 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101510bcc - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x10122a0f4 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101452560 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101aceec0 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101eaa154 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x10285a26c - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x102859bbc - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101580e9c - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1015e7da8 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x220290eb0 - /usr/lib/libnetwork.dylib : <redacted>
	0x21eb34a38 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21eb357d4 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21eae6164 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21eae68d4 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21ed161b4 - /usr/lib/system/libsystem_pthread.dylib : _pthread_wqthread
	0x21ed18cd4 - /usr/lib/system/libsystem_pthread.dylib : start_wqthread

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x10292c49c):0x10292c48c  c0 03 5f d6 ff c3 00 d1 fd 7b 02 a9 fd 83 00 91  .._......{......
0x10292c49c  08 20 40 b9 c8 00 c8 37 88 01 e0 37 00 00 40 f9  . @....7...7..@.
0x10292c4ac  fd 7b 42 a9 ff c3 00 91 c0 03 5f d6 28 10 00 90  .{B......._.(...
0x10292c4bc  08 39 01 91 e8 0b 00 f9 08 10 00 f0 08 4d 3f 91  .9...........M?.

=================================================================
	Managed Stacktrace:
=================================================================
=================================================================

File: Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+        public bool IsSupported => true;
+
+        public IEnumerable<ConnectionType> ConnectionTypes
+        {
+            get
+            {
+                var statuses = _connectionStatuses.Where(x => x.Value).Select(x => x.Key).ToList();
+
+                if(statuses.Contains(NWInterfaceType.Other) && statuses.Count() > 1)
```
Comment (wcoder) [inline Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: Please use `.Count` list property.

File: Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                        case NWInterfaceType.Cellular:
+                            return ConnectionType.Cellular;
+                        case NWInterfaceType.Wifi:
+                            return ConnectionType.WiFi;
+                        default:
+                            return ConnectionType.Other;
+                    }
+                }).ToList();
+
+            if(types.Contains(ConnectionType.Other) && types.Count > 1)
```
Comment (wcoder) [inline Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: Please apply format for fix `if(`

File: Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            {
+                types.Remove(ConnectionType.Other);
```
Comment (wcoder) [inline Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs]: Duplication of check `ConnectionType.Other` and `NWInterfaceType.Other` ?

**Pull request merged**